### PR TITLE
Add Pathauto with default patterns and node fallback

### LINKED ISSUE #153: Explore pathauto defaults.
https://www.drupal.org/project/pathauto

## What is the problem that we want to solve?
In Quickstart 1.0 we use pathauto, and I would like to see this in QS 2.0

### PR DESCRIPTION
**Pull request by joshuasosa**
## Description

Fixes #153 by adding a dependency on pathauto and setting up default patterns for content, taxonomy terms, and users. Patterns are the same as in UAQS. The fallback for empty titles was also ported from UAQS.

## How Has This Been Tested?

Tested locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart project in the right column.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by joshuasosa** [comment]
Tests on demo site:
https://ef460d83-653e-4501-9d91-b5dcdb91b2e6.probo.build/test-1
https://ef460d83-653e-4501-9d91-b5dcdb91b2e6.probo.build/test-2
https://ef460d83-653e-4501-9d91-b5dcdb91b2e6.probo.build/test-2/test-3
https://ef460d83-653e-4501-9d91-b5dcdb91b2e6.probo.build/test-2/test-3/test-4

File: modules/custom/az_core/az_core.module [hunk 1]
```diff
@@ -22,3 +25,37 @@ function az_core_help($route_name, RouteMatchInterface $route_match) {
     default:
   }
 }
+
+/**
+ * Implements hook_pathauto_pattern_alter().
```
Comment (joshuasosa) [inline modules/custom/az_core/az_core.module:30]: This section generally comes from what was is in UAQS. The $alias part was my attempt at doing what pathauto does in [PathautoGenerator](https://git.drupalcode.org/project/pathauto/-/blob/8.x-1.x/src/PathautoGenerator.php#L223). Review of this part would be helpful, especially if something should be handled differently.

**Comment by joeparsons** [comment]
I started to look into what is causing this PHPunit test failure but didn't get very far with it yet:
```
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.11-4+ubuntu16.04.1+deb.sury.org+1
Configuration: /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/phpunit.xml.dist

......E                                                             7 / 7 (100%)

Time: 5.68 minutes, Memory: 8.00MB

There was 1 error:

1) Drupal\Tests\az_layouts\Functional\AZLayoutsUITest::testAddCustomInlineBlock
Drupal\Core\Entity\EntityStorageException: Undefined property: Drupal\pathauto\AliasCleaner::$cleanAlias

/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php:846
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityBase.php:395
/src/az-quickstart-scaffolding/web/core/modules/node/tests/src/Traits/NodeCreationTrait.php:98
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_layouts/tests/src/Functional/AZLayoutsUITest.php:73

Caused by
PHPUnit\Framework\Error\Notice: Undefined property: Drupal\pathauto\AliasCleaner::$cleanAlias

/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_core/az_core.module:48
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleHandler.php:539
/src/az-quickstart-scaffolding/web/modules/contrib/pathauto/src/PathautoGenerator.php:202
/src/az-quickstart-scaffolding/web/modules/contrib/pathauto/src/PathautoGenerator.php:380
/src/az-quickstart-scaffolding/web/modules/contrib/pathauto/pathauto.module:86
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleHandler.php:403
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:206
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php:843
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:535
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php:728
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:460
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php:837
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityBase.php:395
/src/az-quickstart-scaffolding/web/core/modules/node/tests/src/Traits/NodeCreationTrait.php:98
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_layouts/tests/src/Functional/AZLayoutsUITest.php:73

ERRORS!
Tests: 7, Assertions: 101, Errors: 1.
```

**Comment by joshuasosa** [comment]
For this particular default content pattern that relies on the node being in the menu, instead of testing if the alias comes out empty, people seem to like testing if the node actually has menu enabled. The result appears to be the same and resolves the unit test error it was encountering.

**Comment by joeparsons** [comment]
Based on discussion in our July 8 Wednesday PR meeting, it looks like this PR branch might need to merge the latest changes from the master branch because as-is it is using the master branch of the scaffolding repo to build Quickstart and that branch no longer contains the PHPCS config files (they were moved to the profile repo).

**Comment by joshuasosa** [comment]
> Based on discussion in our July 8 Wednesday PR meeting, it looks like this PR branch might need to merge the latest changes from the master branch because as-is it is using the master branch of the scaffolding repo to build Quickstart and that branch no longer contains the PHPCS config files (they were moved to the profile repo).

I did the following:
```
git fetch --all
git reset --hard origin/master
git pull
git push
```
If that looks ok, it should be ready for review again. Otherwise, I can try something else.

**Comment by joeparsons** [comment]
@joshuasosa The diff between your branch and master looks good to me here: https://github.com/az-digital/az_quickstart/compare/feature/153 as well as when I check out your branch locally.  For some reason the "files changed" tab on this PR still seems to be showing additional changes that already exist in master but I think it'd still be safe to merge (especially because we'll do a squash merge).  Thanks.

**Pull request merged**